eZ/Publish/Core/Limitation/ObjectStateLimitationType.php

```diff
@@ -236,8 +236,49 @@ public function getCriterion(APILimitationValue $value, APIUserReference $curren
             return new Criterion\ObjectStateId($value->limitationValues[0]);
         }
 
-        // several limitation values: IN operation
-        return new Criterion\ObjectStateId($value->limitationValues);
+        $groupedLimitationValues = $this->groupLimitationValues($value->limitationValues);
+
+        if (count($groupedLimitationValues) === 1) {
+            // several limitation values: IN operation
+            return new Criterion\ObjectStateId($value->limitationValues);
```

Comment:
Should this be `$groupedLimitationValues[0]` in case some of the filtering being done on that one group? Otherwise maybe comment on why we don't use the value here, if so could the count have been done on`$value->limitationValues)` ?